tests/test_scripts/test_solph/test_tsam/test_storage_invest_tsam_integration.py

```diff
@@ -0,0 +1,216 @@
+# -*- coding: utf-8 -*-
+
+r"""
+General description:
+---------------------
+
+The example models the following energy system:
+
+                input/output   bel
+                     |          |
+ wind(FixedSource)   |--------->|
+                     |          |
+ demand(Sink)        |<---------|
+                     |          |
+ storage(Storage)    |<---------|
+                     |--------->|
+
+
+
+An initial SOC of zero leads to infeasible solution, as last inter SOC has to
+match first inter SOC.
+Following equations have to be fulfilled:
+:math:`F_{el,st}[0] = F_{el,st}[6]`
+:math:`SOC_{init} * discharge + F_{el,st}[0] =`
+:math:`\sum_{i=1}^{n=5}F_{st,el}[i]/eff_{out}/(1 - discharge)^i`
+:math:`F_{el,st}[6] = (SOC_{init} + F_{el,st}[5]/eff_{out}) / (1 - discharge)`
+
+This file is part of project oemof (github.com/oemof/oemof). It's copyrighted
+by the contributors recorded in the version control history of the file,
+available from its original location oemof/tests/test_scripts/test_solph/
+test_storage_investment/test_storage_investment.py
+
+SPDX-License-Identifier: MIT
+"""
+
+import logging
+
+import pandas as pd
+import pytest
+from oemof.tools import economics
+from oemof.tools import logger
+
+from oemof import solph
+
+##########################################################################
+# Initialize the energy system and read/calculate necessary parameters
+##########################################################################
+
+logger.define_logging()
+logging.info("Initialize the energy system")
+
+tindex_original = pd.date_range("2022-01-01", periods=8, freq="H")
+tindex = pd.date_range("2022-01-01", periods=4, freq="H")
+
+energysystem = solph.EnergySystem(
+    timeindex=tindex,
+    timeincrement=[1] * len(tindex),
+    periods=[tindex],
```

Comment:
this is not needed and should be None, as you dont use multi-period, do you?

Reply:
No, it's the opposite indeed: Currently, TSAM-mode is only available in multi-period mode! But for those examples, I only use a single period to make things easier to test...

Reply:
I still wonder if multi-period should/could be activated ALWAYS, and an "old" default oemof.solph simulation simply uses only a single period... but this is another issue.

Reply:
its basically translated into pyomo like this but a lot of extra constrains and variables are not necessary (e.g. for decommissioning etc)